tests/gis_tests/test_ptr.py

```diff
@@ -0,0 +1,66 @@
+import ctypes
+
+from django.contrib.gis.ptr import CPointerBase
+from django.test import SimpleTestCase, mock
+
+
+class CPointerBaseTest(SimpleTestCase):
+
+    def test(self):
+        destructor_mock = mock.Mock()
+
+        class NullPointerException(Exception):
+            pass
+
+        class FakeGeom1(CPointerBase):
+            null_ptr_exception_class = NullPointerException
+
+        class FakeGeom2(FakeGeom1):
+            ptr_type = ctypes.POINTER(ctypes.c_float)
+            destructor = destructor_mock
+
+        fg1 = FakeGeom1()
+        fg2 = FakeGeom2()
+
+        # These assignments are OK -- None is allowed because
+        # it's equivalent to the NULL pointer.
+        fg1.ptr = fg1.ptr_type()
+        fg1.ptr = None
+        fg2.ptr = fg2.ptr_type(ctypes.c_float(5.23))
+        fg2.ptr = None
+
+        # Because pointers have been set to NULL, an exception should be
+        # raised when we try to access it.  Raising an exception is
+        # preferable to a segmentation fault that commonly occurs when
+        # a C method is given a NULL memory reference.
+        for fg in (fg1, fg2):
+            with self.assertRaises(NullPointerException):
+                fg.ptr
+
+        # Anything that is either not None or the acceptable pointer type will
+        # result in a TypeError when trying to assign it to the `ptr` property.
+        # Thus, memory addresses (integers) and pointers of the incorrect type
+        # (in `bad_ptrs`) will not be allowed.
+        bad_ptrs = (5, ctypes.c_char_p(b'foobar'))
+        for bad_ptr in bad_ptrs:
+            for fg in (fg1, fg2):
+                with self.assertRaises(TypeError):
```

Comment:
Do we have a preference for `assertRaisesMessage` over `assertRaisesRegex`? If so this probably should be documented.